From 443614f91092ef713eb2dd37706c519427c7b120 Mon Sep 17 00:00:00 2001
From: chinarulezzz <drop@chinarulezzz.fun>
Date: Thu, 5 Mar 2020 01:26:31 +0200
Subject: [PATCH] airodump-ng.c: add manufacturer column to the client list

Be aware!

According to this thread:
>> https://github.com/aircrack-ng/aircrack-ng/issues/1023#issuecomment-372062743

this patch is not needed at all :=D

Patch on top of b9ee19bcf868d4f4f0197cb0d886d0ebe91a6636 (git)
---
 src/airodump-ng/airodump-ng.c | 24 +++++++++++++++++++++++-
 1 file changed, 23 insertions(+), 1 deletion(-)

diff --git a/src/airodump-ng/airodump-ng.c b/src/airodump-ng/airodump-ng.c
index 68353fb..0afb607 100644
--- a/src/airodump-ng/airodump-ng.c
+++ b/src/airodump-ng/airodump-ng.c
@@ -4044,9 +4044,17 @@ static void dump_print(int ws_row, int ws_col, int if_num)
 
 	if (lopt.show_sta)
 	{
-		strcpy(strbuf,
+		if (lopt.show_manufacturer)
+			strcpy(strbuf,
+			   " BSSID              STATION "
+			   "           PWR   Rate    Lost    Frames  Notes  Manufacturer                   "
+			   " Probes");
+		else
+			strcpy(strbuf,
 			   " BSSID              STATION "
 			   "           PWR   Rate    Lost    Frames  Notes  Probes");
+
+
 		strbuf[ws_col - 1] = '\0';
 		console_puts(strbuf);
 		CHECK_END_OF_SCREEN();
@@ -4151,6 +4159,20 @@ static void dump_print(int ws_row, int ws_col, int if_num)
 						   ? "PMKID"
 						   : (st_cur->wpa.state == 7 ? "EAPOL" : ""));
 
+				if (lopt.show_manufacturer)
+				{
+					if (st_cur->manuf == NULL)
+					{
+						st_cur->manuf = get_manufacturer(st_cur->stmac[0],
+										 st_cur->stmac[1],
+										 st_cur->stmac[2]);
+
+					}
+					if (strstr(st_cur->manuf, "Unknown") == NULL)
+						printf(strlen(st_cur->manuf) > 32 ? "  %s" : "  %-32s",
+								st_cur->manuf);
+				}
+
 				if (ws_col > (columns_sta - 6))
 				{
 					memset(ssid_list, 0, sizeof(ssid_list));
-- 
2.25.0

